feat(api): PATCH Event Type V2 API to support all current locations

dhairyashiil opened the pull request:
## Video:

https://github.com/user-attachments/assets/60e8bc22-238f-48e0-b143-c37bb432c64a




## What does this PR do?

Fixes validation for integration locations in API v2 event-types endpoints by allowing all 29 supported video conferencing integrations instead of only 4.

API v2 rejected requests with integration locations using integrations other than cal-video, google-meet, office365-video, or zoom with:

```
Validation failed for integration location: integration must be one of the following values: cal-video, google-meet, office365-video, zoom
```
This blocked integrations like whereby-video, whatsapp-video, webex-video, jitsi, huddle, etc., even though they are supported elsewhere.

### Root Cause
- The supportedIntegrations array in the event-types input validation only listed 4 integrations, while:
- The transformer mappings (api-to-internal and internal-to-api) support 29 integrations
- The bookings input validation already includes all 29
- The output types include all 29
- The app store configs define all 29
- This mismatch caused validation to reject valid integrations.

### Solution
Updated supportedIntegrations in packages/platform/types/event-types/event-types_2024_06_14/inputs/locations.input.ts to include all 29 supported integrations, matching:
- The transformer mappings
- The bookings input validation
- The output types

dhairyashiil committed:
feat(api): PATCH Event Type V2 API to support all current locations

dhairyashiil committed:
docs(api): update locations documentation and add E2E tests for new integrations

- Updated locations property documentation in create-event-type.input.ts
  and update-event-type.input.ts to clarify app installation requirements
- Explained that only Google Meet, MS Teams, and Zoom can be installed via API
- Noted that Cal Video is installed by default
- Added E2E tests for creating and updating event types with newly supported
  integration locations (jitsi, zoom, google-meet, whereby, huddle, element-call)
- Regenerated openapi.json with updated API documentation

Addresses feedback from Lauris regarding platform API location support.

dhairyashiil committed:
Merge remote-tracking branch 'upstream/main' into feat/v2-update-event-type-support-all-current-locations

dhairyashiil committed:
fix(api): use supportedIntegrations list for app validation

Updated checkAppIsValidAndConnected to use the full supportedIntegrations
list from locations.input.ts instead of hardcoded array. This allows all
27 supported conferencing apps to be set as event type locations via API,
as long as they are already connected by the user.

dhairyashiil committed:
fix(api): add slug mapping for all conferencing integrations

Added comprehensive slug mapping to translate API integration names
(e.g., 'facetime-video', 'whereby-video') to actual app slugs
(e.g., 'facetime', 'whereby'). This ensures the app lookup works
correctly for all 27 supported conferencing integrations.

Addresses AI bot feedback about slug mismatches.

dhairyashiil committed:
fix(api): add missing huddle to huddle01 slug mapping

Added mapping for huddle -> huddle01. Other apps like tandem, jitsi,
cal-video, google-meet, and zoom don't need mapping as their API names
already match their app slugs (handled by the fallback || appSlug).

dhairyashiil commented on apps/api/v2/src/ee/event-types/event-types_2024_06_14/services/input-event-types.service.ts:
```diff
@@ -542,14 +543,39 @@ export class InputEventTypesService_2024_06_14 {
-      appSlug = "msteams";
-    }
+    // Map API integration names to actual app slugs
+    const slugMap: Record<string, string> = {
```
huddle needed mapping, but tandem and jitsi didn't because the fallback || appSlug handles matching names

dhairyashiil committed:
update key

dhairyashiil committed:
update ket

dhairyashiil committed:
Merge branch 'main' into feat/v2-update-event-type-support-all-current-locations

dhairyashiil committed:
test(api): update E2E tests to validate newly supported integrations

Replaced end-to-end tests with validation-focused tests that follow
the existing pattern. The new test creates event types with various
newly supported integrations (jitsi, whereby-video, huddle, tandem,
element-call-video) directly in the database (bypassing app connection
checks) and verifies the API correctly returns them.

This approach tests that the input validation accepts all 27 supported
integration types without requiring actual app installations in the
test environment.

dhairyashiil committed:
Merge branch 'feat/v2-update-event-type-support-all-current-locations' of https://github.com/dhairyashiil/cal.com into feat/v2-update-event-type-support-all-current-locations

pedroccastro commented on apps/api/v2/src/ee/event-types/event-types_2024_06_14/services/input-event-types.service.ts:
```diff
@@ -542,14 +543,40 @@ export class InputEventTypesService_2024_06_14 {
-      appSlug = "msteams";
-    }
+    // Map API integration names to actual app slugs
+    const slugMap: Record<string, string> = {
```
Hey @dhairyashiil! Nice work expanding the supported integrations!

Found a few slug mismatches while testing:

- `"whatsapp-video"` should map to `"whatsapp"` (currently `"whatsappvideo"`)
- `"shimmer-video"` should map to `"shimmervideo"` (currently `"shimmer"`)
- `"jelly-conferencing"` should map to `"jelly"` (currently `"jelly-conferencing"`)

Tested `shimmer-video` with the app connected and got "shimmer not connected" error.
Correct slugs are in `packages/app-store/{whatsapp,shimmervideo,jelly}/config.json`.

dhairyashiil committed:
Merge remote-tracking branch 'upstream/main' into feat/v2-update-event-type-support-all-current-locations

dhairyashiil committed:
fix(api): correct slug mappings for whatsapp, shimmer, and jelly integrations

- Fixed whatsapp-video mapping from 'whatsappvideo' to 'whatsapp'
- Fixed shimmer-video mapping from 'shimmer' to 'shimmervideo'
- Fixed jelly-conferencing mapping from 'jelly-conferencing' to 'jelly'

All slug mappings now correctly match the actual app slugs in
packages/app-store/*/config.json files. This ensures proper app
validation when users create/update event types with these locations.

Addresses feedback from @pedroccastro

dhairyashiil committed:
updated openapi.json file because of main branch code

supalarry commented on apps/api/v2/src/ee/event-types/event-types_2024_06_14/controllers/event-types.controller.e2e-spec.ts:
```diff
@@ -2855,6 +2855,57 @@ describe("Event types Endpoints", () => {
         });
     });
 
+    it("should accept newly supported integration locations in input validation", async () => {
```
Great job on the tests! Can we also add a negative test aka that if we create event type with non jitsi location via the event types fixture that we get the 400 error in the loom you created and expect that we have that exact error message like we do elsewhere:
<img width="1192" height="407" alt="Screenshot 2025-11-27 at 10 05 28" src="https://github.com/user-attachments/assets/98faf882-8f91-4a23-852e-5c02863c3b36" />

dhairyashiil replied:
done, video:


https://github.com/user-attachments/assets/c4ea0231-5bd9-4ff0-9f32-eb4e863bca3a

supalarry reviewed (commented):
Great job! Have just 1 comment:

dhairyashiil committed:
test(api): add negative test for unsupported integration locations

- Added E2E test to validate 400 error when creating event type with unsupported integration
- Test verifies exact error message listing all supported integrations
- Uses imported supportedIntegrations constant for maintainability
- Follows same pattern as booking fields validation tests

Addresses feedback from @supalarry

dhairyashiil committed:
Merge remote-tracking branch 'upstream/main' into feat/v2-update-event-type-support-all-current-locations

supalarry reviewed (commented):
Thanks for adding a new test.

Sorry if i wasn't clear - the test you added is good to have, but the test tests creating a new event type, but this PR is about patching it and i wanted that you add a patch test that tries to connect jitsi to event type that does not have jitsi installed so the error is "jitsi is not connected". Does that make sense?

dhairyashiil committed:
test(api): add negative test for patching event type with unconnected integration

- Added E2E test to validate 400 error when user tries to PATCH event type with jitsi integration they haven't connected
- Test verifies exact error message 'jitsi not connected.'
- Follows existing test patterns with proper cleanup

dhairyashiil committed:
Merge remote-tracking branch 'upstream/main' into feat/v2-update-event-type-support-all-current-locations

supalarry reviewed (approved):
Nice PR !
